control/pid_longitudinal_controller/src/longitudinal_controller_utils.cpp

```diff
@@ -168,5 +168,39 @@ double applyDiffLimitFilter(
   const double min_val = -max_val;
   return applyDiffLimitFilter(input_val, prev_val, dt, max_val, min_val);
 }
+
+geometry_msgs::msg::Pose findTrajectoryPoseAfterDistance(
+  const size_t src_idx, const double distance,
+  const autoware_auto_planning_msgs::msg::Trajectory & trajectory)
+{
+  double remain_dist = distance;
+  geometry_msgs::msg::Pose p = trajectory.points.back().pose;
+  for (size_t i = src_idx; i < trajectory.points.size() - 1; ++i) {
+    const double dist = tier4_autoware_utils::calcDistance3d(
+      trajectory.points.at(i).pose, trajectory.points.at(i + 1).pose);
+    if (remain_dist < dist) {
+      if (remain_dist <= 0.0) {
+        return trajectory.points.at(i).pose;
+      }
+      double ratio = remain_dist / dist;
+      p = trajectory.points.at(i).pose;
+      p.position.x = interpolation::lerp(
+        trajectory.points.at(i).pose.position.x, trajectory.points.at(i + 1).pose.position.x,
+        ratio);
+      p.position.y = interpolation::lerp(
+        trajectory.points.at(i).pose.position.y, trajectory.points.at(i + 1).pose.position.y,
+        ratio);
+      p.position.z = interpolation::lerp(
+        trajectory.points.at(i).pose.position.z, trajectory.points.at(i + 1).pose.position.z,
+        ratio);
+      p.orientation = lerpOrientation(
+        trajectory.points.at(i).pose.orientation, trajectory.points.at(i + 1).pose.orientation,
+        ratio);
```

Comment:
This would be more simple?

```suggestion
      const auto p0 = trajectory.points.at(i).pose;
      const auto p1 = trajectory.points.at(i).pose;
      p.position.x = interpolation::lerp(p0.position.x, p1.position.x, ratio);
      p.position.y = interpolation::lerp(p0.position.y, p1.position.y, ratio);
      p.position.z = interpolation::lerp(p0.position.z, p1.position.z, ratio);
      p.orientation = lerpOrientation(p0.orientation, p1.orientation, ratio);
```